src/tcp_act.c

```diff
@@ -686,6 +720,54 @@ static enum act_parse_ret tcp_parse_silent_drop(const char **args, int *cur_arg,
 	return ACT_RET_PRS_OK;
 }
 
+/* Parse a "set-tlv" action. It takes the TLV ID as argument which can be set to a
+ * sample expression. It returns ACT_RET_PRS_OK on success, ACT_RET_PRS_ERR on error.
+ */
+static enum act_parse_ret tcp_parse_set_tlv(const char **args, int *cur_arg, struct proxy *px,
+		struct act_rule *rule, char **err)
+{
+	const char *cmd_name = args[*cur_arg-1];
+	struct sample_expr *expr;
+
+	char *error;
+	unsigned int tlv_id;
+
+	expr = sample_parse_expr((char **)args, cur_arg, px->conf.args.file, px->conf.args.line, err, &px->conf.args, NULL);
+	if (!expr || !expr->fetch->val)
+		return ACT_RET_PRS_ERR;
+
+	cmd_name += strlen("set-tlv");
+
+	if (*cmd_name == '(') {
+		cmd_name++; /* skip the '(' */
+		tlv_id = strtoul(cmd_name, &error, 0); /* convert TLV ID */
```

Comment:
If there are no valid digits, 0 is returned and `errno` is set to `EINVAL`. However, currently, it would be considered as valid input.